File: Binary_Search /python/1920.py
n = int(input())
n_arr = sorted(list(map(int,input().split())))
m = int(input())
m_arr = list(map(int,input().split()))

def search(min_idx,max_idx):
    if min_idx>max_idx:
        return 0
    half_idx = min_idx + (max_idx-min_idx)//2
    if n_arr[half_idx]>m_arr[idx]:
        result = search(min_idx,half_idx-1)
    elif n_arr[half_idx] == m_arr[idx]:
        return 1
    else:
        result = search(half_idx+1,max_idx)
    return result
result = []
for idx in range(m):
    result.append(search(0,n-1))
print("\n".join(map(str,result)))

File: Binary_Search /javascript/1920.js
const input = require('fs').readFileSync('/dev/stdin').toString().split('\n');
let [nArr,mArr] = [input[1].split(' ').map(Number).sort((a,b)=>a-b),input[3].split(' ').map(Number)];

const search = (minIdx,maxIdx) =>{
    if (minIdx > maxIdx) return 0
    const halfIdx = minIdx + Math.floor((maxIdx-minIdx) /2);
    if (nArr[halfIdx]> mArr[idx]){
      result = search(minIdx,halfIdx-1);
    }
    else if (nArr[halfIdx] === mArr[idx]){
      return 1;
    }
    else{
      result = search(halfIdx+1,maxIdx);
    }
    return result;
}
let idx = -1;
console.log((mArr.reduce((acc,cur) => {idx++; acc.push(search(0,nArr.length-1)); return acc} ,[])).join("\n"));
